tox.ini

```diff
@@ -36,10 +36,17 @@ deps = sphinx
 commands = sphinx-build -b html -d docs/_build/doctrees docs docs/_build/html
 whitelist_externals = make
 
-[testenv:py37-flake8]
+[testenv:py37-lint]
 description = Static code analysis and code style
-deps = flake8
-commands = flake8
+skip_install = True
+deps =
+    flake8
+    isort
+    black
+commands =
+    flake8
+    isort tests setup.py analytical --check --diff
+    black tests setup.py analytical --check
```

Comment:
I'm not a big fan of combining several checks or tests:

If the `py37-lint` job fails will you know what (probably) happened without having to take a closer look?

It's like with writing tests: A test should ideally test a single thing.

Reply:
Ok -- I've split them into their own environments. 

I have a slightly different view in that I'd take the speed benefit of not creating environments over having to click on the env to find out what failed. But that's ok, this way works just fine as well.

Reply:
That's true, what you say. But then again, the 3 jobs are run in parallel on Microsoft's infrastructure, and that's amazingly fast. The benefit of saving seconds of my attention span is much more valuable then.

Also, take a look at #174 for the "next generation" of splitting up jobs. This will make everything much more transparent and easier to grasp, whether things go wrong or go well (think of new volunteers joining the project).

Every problem should have its own, dedicated indicator. After that, our goal (as an industry) needs to be to make the overhead of the single job so minimal that the difference between running them in a block or separate is almost unnoticeable. :alien: